pkg/engine/planner/physical/optimizer.go

```diff
@@ -198,26 +199,54 @@ func (r *projectionPushdown) apply(node Node) bool {
 		// Always project timestamp column
 		projections[len(node.PartitionBy)] = &ColumnExpr{Ref: types.ColumnRef{Column: types.ColumnNameBuiltinTimestamp, Type: types.ColumnTypeBuiltin}}
 
-		return r.applyProjectionPushdown(node, projections, false)
+		ambiguousProjections := collectAmbiguousColumns(node.PartitionBy)
+		return r.applyProjectionPushdown(node, projections, ambiguousProjections, false)
+	case *VectorAggregation:
+		if len(node.GroupBy) == 0 || node.Operation != types.VectorAggregationTypeSum {
```

Comment:
I guess this is copied from the RangeAggregation, but I'm not sure why we only allow it to pushdown predicates for Count/Sum and none of the others? 

cc @ashwanthgoli

Reply:
we have a check earlier in the pipeline that returns an unimplemented for anything other than sum by count_over_time, so my assumption was this was a byproduct of that, and meant to be temporary until we implement it?

that being said, in order to support drilldown we'll need to support avg_over_time, which is just a sum/count, so since we already have count I think we should implement sum next.